### PR TITLE
Feature/offline config

### PR DESCRIPTION
**Pull request by ochococo**
In `DEBUG` builds, we load local configuration file.

### CODE REVIEW

File: GrandCentralBoard/App/AppDelegate.swift [hunk 1]
```diff
@@ -5,6 +5,10 @@
 
 import UIKit
 
+let debugBuild: Bool = _isDebugAssertConfiguration()
```
Comment (krris) [inline GrandCentralBoard/App/AppDelegate.swift]: Isn't it private API?

File: GrandCentralBoard/App/ViewControllers/LoadingViewController.swift [hunk 2]
```diff
@@ -6,13 +6,20 @@
...
+private let availableBuilders: [WidgetBuilding] = [WatchWidgetBuilder(dataDownloader: dataDownloader), BonusWidgetBuilder(dataDownloader: dataDownloader)]
+private let useLocal = debugBuild
 
 class LoadingViewController: UIViewController {
 
-    let configurationPath = "http://gcb.macoscope.com/configuration.json"
-    let configurationDownloader = ConfigurationDownloader(dataDownloader: DataDownloader())
+    lazy var configurationFetching: ConfigurationFetching = {
+
+        if useLocal {
```
Comment (krris) [inline GrandCentralBoard/App/ViewControllers/LoadingViewController.swift]: No it uses local configuration on `release` build . It should use it on `debug` build.
screen shot 2016-04-01 at 08 36 07

File: GrandCentralBoard/Core/Configuration/ConfigurationDownloader.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        self.builders = builders
+        self.dataDownloader = dataDownloader
+    }
+
+    public func fetchConfiguration(closure: (Result<Configuration>) -> ()) {
+        dataDownloader.downloadDataAtPath(path) { [weak self] result in
+            switch result {
+                case .Success(let data):
+                    do {
+                        guard let instance = self else { return }
```
Comment (krris) [inline GrandCentralBoard/Core/Configuration/ConfigurationDownloader.swift]: If `instance != self` shouldn't we call a completion block?

**Comment by krris** [comment]
:tennis: @ochococo

File: GrandCentralBoard/Core/Configuration/ConfigurationDownloader.swift (hunk 3 above)
Comment (ochococo) [inline GrandCentralBoard/Core/Configuration/ConfigurationDownloader.swift]: No. We're dead.

File: GrandCentralBoard/App/AppDelegate.swift (hunk 1 above)
Comment (ochococo) [inline GrandCentralBoard/App/AppDelegate.swift]: giphy

Comment (ochococo) [inline GrandCentralBoard/App/AppDelegate.swift]: But seriously - it's not for the AppStore for now.

File: GrandCentralBoard/App/ViewControllers/LoadingViewController.swift (hunk 2 above)
Comment (ochococo) [inline GrandCentralBoard/App/ViewControllers/LoadingViewController.swift]: Cannot reproduce.

Comment (krris) [inline GrandCentralBoard/App/ViewControllers/LoadingViewController.swift]: Oh, I had to reset simulator and it started to work.

**Comment by krris** [comment]
Good job! I like this improvement :clap:

**Pull request merged**